tests/integration/route_test.py

```diff
@@ -436,3 +442,48 @@ def test_iface_down_with_routes_in_current(eth1_up, get_routes_func):
 
     cur_state = libnmstate.show()
     _assert_routes([], cur_state)
+
+
+@pytest.fixture(scope='function')
+def eth1_static_gateway_dns(eth1_up):
+    routes = (
+        [_get_ipv4_gateways()[0], _get_ipv6_gateways()[0]]
+        + _get_ipv4_test_routes()
+        + _get_ipv6_test_routes()
+    )
+
+    state = {
+        Interface.KEY: [ETH1_INTERFACE_STATE],
+        Route.KEY: {Route.CONFIG: routes},
+        DNS.KEY: {
+            DNS.CONFIG: {
+                DNS.SERVER: [IPV6_DNS_NAMESERVER, IPV4_DNS_NAMESERVER],
+                DNS.SEARCH: DNS_SEARCHES,
+            }
+        },
+    }
+
+    libnmstate.apply(state)
+    yield state
+    # Remove DNS config
+    libnmstate.apply(
+        {Interface.KEY: [], DNS.KEY: {DNS.CONFIG: {}}}, verify_change=False
+    )
+
+
+@pytest.mark.xfail(
+    raises=AssertionError,
+    reason='https://bugzilla.redhat.com/1748389',
+    strict=True,
+)
+def test_apply_empty_state_could_preserve_routes(eth1_static_gateway_dns):
```

Comment:
Please drop `could`.

What about the DNS entries? They are also wiped out, right?

Reply:
The DNS entry is OK.